dags/populate_metabase_fluxiae.py

```diff
@@ -0,0 +1,117 @@
+import glob
+import logging
+import shutil
+import subprocess
+import tempfile
+
+import airflow
+from airflow.decorators import task
+from airflow.models import Variable
+from airflow.operators.trigger_dagrun import TriggerDagRunOperator
+
+from dags.common import default_dag_args, s3, slack
+from dags.common.flux_iae import get_fluxiae_df, get_fluxiae_referential_filenames, store_df
+
+
+logger = logging.getLogger(__name__)
+
+with airflow.DAG(
+    **default_dag_args(),
+    dag_id="populate_metabase_fluxiae",
+    schedule="0 13 * * 1",
+) as dag:
+
+    @task(task_id="create_work_directory")
+    def create_work_directory(*, task_instance, **kwargs):
+        dag_run = task_instance.get_dagrun()
+        return tempfile.mkdtemp(prefix=f"{dag_run.dag_id}_", suffix=f"_{dag_run.run_id}")
+
+    @task(task_id="import_and_decrypt")
+    def import_and_decrypt(import_directory, **kwargs):
+        bucket_name = Variable.get("DATASTORE_S3_EMPLOIS_BUCKET_NAME")
+        client = s3.client()
+
+        # Use the most recent file available in the bucket
+        response = client.list_objects_v2(Bucket=bucket_name, Prefix="flux-iae/")
+        if not response["KeyCount"]:
+            raise RuntimeError("No files found in bucket")
+
+        with tempfile.NamedTemporaryFile(dir=import_directory) as file:
+            most_recent_import = max(obj["Key"] for obj in response["Contents"])
+            logger.info("Most recent file is: %r", most_recent_import)
+
+            # Check that the file is new; this DAG shouldn't run on the same file twice.
+            if Variable.get("ASP_FLUX_IAE_MOST_RECENT_IMPORT", None) == most_recent_import:
+                raise RuntimeError(
+                    f"{most_recent_import} is the most recent FluxIAE import in the bucket, "
+                    "but a file with the same key was already imported! "
+                    "A new export may not have uploaded to the bucket. To re-run the DAG with the file, "
+                    "clear the value of ASP_FLUX_IAE_MOST_RECENT_IMPORT from the Airflow variables."
+                )
+
+            # Retrieve the encrypted and compressed file from S3 storage.
+            client.download_fileobj(bucket_name, most_recent_import, file)
+
+            # Unzip the file.
+            # NOTE: tar.gz extension but it is a Zip file...
+            logger.info("Decompressing file %r into %r", file.name, import_directory)
+            subprocess.check_call(["7z", "x", file.name, f"-o{import_directory}"])
+
+        # Decrypt the contents.
+        for aes_encrypted_csv in glob.glob(f"{import_directory}/*.csv.gz"):
+            logger.info("Decompressing file %r into %r", aes_encrypted_csv, import_directory)
+            password = Variable.get("ASP_RIAE_UNZIP_PASSWORD")
+            subprocess.check_call(["7z", "x", f"-p{password}", aes_encrypted_csv, f"-o{import_directory}"])
```

Comment:
Dernier truc qui me chafouine, dans la version actuelle on se retrouve à passer le mot de passe en clair (ce qui fait qu'on l'affiche en cas d'erreur :face_with_diagonal_mouth:), c'est pas top donc je te propose de faire comme ceci :
```suggestion
            subprocess.check_call(
                " ".join(["7z", "x", "-p${ASP_RIAE_UNZIP_PASSWORD}", aes_encrypted_csv, f"-o{import_directory}"]),
                env={"ASP_RIAE_UNZIP_PASSWORD": Variable.get("ASP_RIAE_UNZIP_PASSWORD").encode()},
                shell=True,
            )
```
Le `" ".join(...)` est nécessaire à cause du `shell=True` qui est lui même nécessaire à l'expansion de la variable `ASP_RIAE_UNZIP_PASSWORD` depuis l’environnement.